### PR TITLE
Added AuthorizedMintRep scheme

### LINKED ISSUE #703: add AuthorizedMintRep scheme
Needed for ETHDenver dao.
A scheme which whitelist an account for minting rep.
initialised with:
-  activationStartTime
-  activationEndTime
-  total reputation reward allowed (0 value will allow infinite reputation reward).
 method :
 - reputationMint(address, amount) onlyAuthorized.

### PR DESCRIPTION
**Pull request by ben-kaufman**
Fix #703

### CODE REVIEW

File: contracts/schemes/AuthorizedMintRep.sol [hunk 1]
```diff
@@ -0,0 +1,75 @@
+pragma solidity 0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../controller/Controller.sol";
+
+/**
+ * @title A scheme for reputation allocation according to token balances
```
Comment (orenyodfat) [inline contracts/schemes/AuthorizedMintRep.sol]: this is not right.

File: contracts/schemes/AuthorizedMintRep.sol [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    uint256 public activationStartTime;
+    uint256 public activationEndTime;
+    uint256 public repRewardLeft;
+    address public authorizedAddress;
+    bool public limitRepReward;
+
+    /**
+     * @dev Throws if called by an unauthorized account.
+     */
+    modifier onlyAuthorized() {
```
Comment (orenyodfat) [inline contracts/schemes/AuthorizedMintRep.sol]: how about use ownable ?

Comment (orenyodfat) [reply contracts/schemes/AuthorizedMintRep.sol]: and onlyOwner

File: contracts/schemes/AuthorizedMintRep.sol (hunk 1 above)
Comment (ben-kaufman) [reply contracts/schemes/AuthorizedMintRep.sol]: fixed

File: contracts/schemes/AuthorizedMintRep.sol [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    ) external {
+        require(avatar == Avatar(0), "can be called only one time");
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        require(_activationStartTime < _activationEndTime, "_activationStartTime < _activationEndTime");
+        avatar = _avatar;
+        activationStartTime = _activationStartTime;
+        activationEndTime = _activationEndTime;
+        repRewardLeft = _maxRepReward;
+        authorizedAddress = _authorizedAddress;
+        limitRepReward = _maxRepReward != 0;
```
Comment (orenyodfat) [inline contracts/schemes/AuthorizedMintRep.sol]: what does it means != 0 ?

File: contracts/schemes/AuthorizedMintRep.sol [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    /**
+     * @dev reputationMint function
+     * @param _beneficiary the beneficiary address to redeem for
```
Comment (orenyodfat) [inline contracts/schemes/AuthorizedMintRep.sol]: redeem ?

File: contracts/schemes/AuthorizedMintRep.sol [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+     * @param _amount the agreementHash hash
```
Comment (orenyodfat) [inline contracts/schemes/AuthorizedMintRep.sol]: _agreementHash ?

File: contracts/schemes/AuthorizedMintRep.sol [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+     */
+    function reputationMint(address _beneficiary, uint256 _amount) external onlyAuthorized {
```
Comment (orenyodfat) [inline contracts/schemes/AuthorizedMintRep.sol]: does this function should return bool same as the reputation mint ?

File: contracts/schemes/AuthorizedMintRep.sol (hunk 3 above)
Comment (ben-kaufman) [reply contracts/schemes/AuthorizedMintRep.sol]: If it is not 0 I want to skip the check of the limit of reputation

File: contracts/schemes/AuthorizedMintRep.sol (hunk 6 above)
Comment (ben-kaufman) [reply contracts/schemes/AuthorizedMintRep.sol]: No, but it can if we want it to.

File: contracts/schemes/AuthorizedMintRep.sol (hunk 3 above)
Comment (orenyodfat) [reply contracts/schemes/AuthorizedMintRep.sol]: this is strange syntax

Comment (ben-kaufman) [reply contracts/schemes/AuthorizedMintRep.sol]: Yes, what's weird about this syntax?

**Pull request merged**